### PR TITLE
Adds Apache Airflow mixin

### PR DESCRIPTION
**Pull request by StefanKurek**
Adds 1 dashboard and several new alerts for Apache Airflow.

Dashboards:

- Airflow overview

Alerts:

- ApacheAirflowStarvingPoolTasks
- ApacheAirflowDAGScheduleDelayWarningLevel
- ApacheAirflowDAGScheduleDelayCriticalLevel
- ApacheAirflowDAGFailures

Apache Airflow overview
image
image
image

### CODE REVIEW

File: apache-airflow-mixin/alerts/alerts.libsonnet [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                The number of DAG failures seen is {{ printf "%%.0f" $value }} over the last 1m for {{ $labels.instance }} - {{ $labels.dag_id }} which is above the threshold of %(alertsCriticalFailedDAGs)s.
+              ||| % $._config,
+            },
+          },
+        ],
+      },
+    ],
+  },
+}
+
```
Comment (JonathanWamsley) [inline apache-airflow-mixin/alerts/alerts.libsonnet:76]: lint error no new line (I think there are spaces here)
```suggestion

```

File: apache-airflow-mixin/dashboards/airflow-overview.libsonnet [hunk 2]
```diff
@@ -0,0 +1,1156 @@
...
+    enableLogDetails: true,
+    prettifyLogMessage: false,
+    showCommonLabels: false,
+    showLabels: false,
+    showTime: false,
+    sortOrder: 'Descending',
+    wrapLogMessage: false,
+  },
+};
+
```
Comment (JonathanWamsley) [inline apache-airflow-mixin/dashboards/airflow-overview.libsonnet:1021]: 2 empty lines here
```suggestion
```

File: apache-airflow-mixin/README.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  },
+}
+```
+
+In order for the selectors to properly work for system logs ingested into your logs datasource, please also include the matching `instance` and `job` labels onto the [scrape_configs](https://grafana.com/docs/loki/latest/clients/promtail/configuration/#scrape_configs) as to match the labels for ingested metrics. It is also important to add matching `dag_id` and `task_id` for the task logs to match the labels for ingested metrics as well as add labels for `dag_file` for the scheduler logs to allow for filtering in the dashboard.
+
+```yaml
+scrape_configs:
+- job_name: system
+  static_configs:
```
Comment (armstrmi) [inline apache-airflow-mixin/README.md:38]: just wondering, should the `job_name` here match the technology like `apache-airflow`

File: apache-airflow-mixin/dashboards/airflow-overview.libsonnet [hunk 4]
```diff
@@ -0,0 +1,1156 @@
...
+          [
+            template.new(
+              'job',
+              promDatasource,
+              'label_values(airflow_scheduler_tasks_executable,job)',
+              label='Job',
+              refresh=1,
+              includeAll=true,
+              multi=true,
+              allValues='',
```
Comment (JonathanWamsley) [inline apache-airflow-mixin/dashboards/airflow-overview.libsonnet:1065]:
```suggestion
              allValues='.+',
```

File: apache-airflow-mixin/alerts/alerts.libsonnet (hunk 1 above)
Comment (armstrmi) [inline apache-airflow-mixin/alerts/alerts.libsonnet:76]: nit: removed added space on line 76

```suggestion
}

```

File: apache-airflow-mixin/dashboards/airflow-overview.libsonnet [hunk 5]
```diff
@@ -0,0 +1,1156 @@
...
+            ),
+            template.new(
+              'instance',
+              promDatasource,
+              'label_values(airflow_scheduler_tasks_executable{job=~"$job"}, instance)',
+              label='Instance',
+              refresh=1,
+              includeAll=true,
+              multi=true,
+              allValues='',
```
Comment (JonathanWamsley) [inline apache-airflow-mixin/dashboards/airflow-overview.libsonnet:1076]:
```suggestion
              allValues='.+',
```

File: apache-airflow-mixin/dashboards/airflow-overview.libsonnet [hunk 6]
```diff
@@ -0,0 +1,1156 @@
...
+              refresh='load'
+            ),
+          ] else [],
+          [
+            template.new(
+              'job',
+              promDatasource,
+              'label_values(airflow_scheduler_tasks_executable,job)',
+              label='Job',
+              refresh=1,
```
Comment (JonathanWamsley) [inline apache-airflow-mixin/dashboards/airflow-overview.libsonnet:1062]:
```suggestion
              refresh=2,
```

**Review by JonathanWamsley — CHANGES_REQUESTED** [review]
Some minor lint changes

**Review by armstrmi — COMMENTED** [review]
A minor fix or two, other than that in addition to what Jon, looks pretty good

File: apache-airflow-mixin/README.md (hunk 3 above)
Comment (StefanKurek) [reply apache-airflow-mixin/README.md:38]: @armstrmi good catch. i'll make that change

File: apache-airflow-mixin/dashboards/airflow-overview.libsonnet (hunk 6 above)
Comment (StefanKurek) [reply apache-airflow-mixin/dashboards/airflow-overview.libsonnet:1062]: @JonathanWamsley i'm not opposed to this. 2s seems fine vs 1s. but any specific reason why? should we always be doing this across the board as a best practice? did this suggestion come from grafana at some point?

File: apache-airflow-mixin/dashboards/airflow-overview.libsonnet (hunk 4 above)
Comment (StefanKurek) [reply apache-airflow-mixin/dashboards/airflow-overview.libsonnet:1065]: @JonathanWamsley allValues='' seems to work just fine for selectors when All is selected. any specific reason for the suggested change that i'm not quite understanding off the bat?

File: apache-airflow-mixin/dashboards/airflow-overview.libsonnet (hunk 5 above)
Comment (StefanKurek) [reply apache-airflow-mixin/dashboards/airflow-overview.libsonnet:1076]: @JonathanWamsley same comment as above

File: apache-airflow-mixin/alerts/alerts.libsonnet (hunk 1 above)
Comment (StefanKurek) [reply apache-airflow-mixin/alerts/alerts.libsonnet]: Should be fixed

File: apache-airflow-mixin/dashboards/airflow-overview.libsonnet (hunk 2 above)
Comment (StefanKurek) [reply apache-airflow-mixin/dashboards/airflow-overview.libsonnet:1021]: Should be fixed

File: apache-airflow-mixin/README.md (hunk 3 above)
Comment (StefanKurek) [reply apache-airflow-mixin/README.md]: Should be fixed

File: apache-airflow-mixin/alerts/alerts.libsonnet (hunk 1 above)
Comment (StefanKurek) [reply apache-airflow-mixin/alerts/alerts.libsonnet]: Should be fixed

File: apache-airflow-mixin/README.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+| ApacheAirflowDAGFailures                    | There have been DAG failures detected.                                              |
+
+Default thresholds can be configured in `config.libsonnet`
+
+```js
+{
+  _config+:: {
+    alertsCriticalPoolStarvingTasks: 0,
+    alertsWarningDAGScheduleDelayLevel: 10,  //s
+    alertsCriticalDAGScheduleDelayLevel: 60,  //s
```
Comment (armstrmi) [inline apache-airflow-mixin/README.md:89]: Just wondering, why have this as 60s vs 1m?

Comment (StefanKurek) [reply apache-airflow-mixin/README.md:89]: @armstrmi no particular reason

File: apache-airflow-mixin/dashboards/airflow-overview.libsonnet (hunk 4 above)
Comment (JonathanWamsley) [reply apache-airflow-mixin/dashboards/airflow-overview.libsonnet:1064]: Screenshot 2023-05-15 at 1 39 24 PM

Its a lint suggestion that we have done previously on a number of dashboards, but not all of them. 
Screenshot 2023-05-15 at 1 43 47 PM

File: apache-airflow-mixin/dashboards/airflow-overview.libsonnet (hunk 5 above)
Comment (JonathanWamsley) [reply apache-airflow-mixin/dashboards/airflow-overview.libsonnet:1075]: ^ dito

File: apache-airflow-mixin/dashboards/airflow-overview.libsonnet (hunk 6 above)
Comment (JonathanWamsley) [reply apache-airflow-mixin/dashboards/airflow-overview.libsonnet:1061]: another lint suggestion from grafanas dashboard linter. Not all the submitted dashboards have this in though, so it is not strictly enforced as of now.

**Review by JonathanWamsley — APPROVED** [review]

**Review by mshahzeb — APPROVED** [review]
Great looking dashboards! LGTM

**Review by gaantunes — APPROVED** [review]
LGTM.

**Pull request merged**